### ISSUE TITLE
NPE in AbstractGeneralizedAffineTerm.div(AbstractGeneralizedAffineTerm.java:395)

### ISSUE THREAD

**Issue by danieldietsch** [ultimate-pa/ultimate#563]
Version: 1a84f73
File [lcm2_valuebound10.c](https://struebli.informatik.uni-freiburg.de/trunk/examples/svcomp/nla-digbench-scaling/lcm2_valuebound10.c)
Log: [lcm2_valuebound10](https://struebli.informatik.uni-freiburg.de/logs/student-acceleration/20210501-092903-student-acceleration-1a84f7301f/student-acceleration.2021-04-30_22-35-57.logfiles/Automizer%20IcfgTranformation%20Jordan.lcm2_valuebound10.yml.log)

Stacktrace
```
java.lang.NullPointerException
	at java.base/java.util.Arrays.stream(Arrays.java:5614)
	at de.uni_freiburg.informatik.ultimate.lib.smtlibutils.polynomials.AbstractGeneralizedAffineTerm.div(AbstractGeneralizedAffineTerm.java:395)
	at de.uni_freiburg.informatik.ultimate.lib.smtlibutils.polynomials.SolveForSubjectUtils.constructRhsIntegerQuotient(SolveForSubjectUtils.java:456)
	at de.uni_freiburg.informatik.ultimate.lib.smtlibutils.polynomials.ExplicitLhsPolynomialRelation.divideByIntegerCoefficient(ExplicitLhsPolynomialRelation.java:224)
	at de.uni_freiburg.informatik.ultimate.lib.smtlibutils.polynomials.SolveForSubjectUtils.solveForSubjectWithoutTreatableDivMod(SolveForSubjectUtils.java:110)
	at de.uni_freiburg.informatik.ultimate.lib.smtlibutils.polynomials.SolveForSubjectUtils.solveForSubject(SolveForSubjectUtils.java:82)
	at de.uni_freiburg.informatik.ultimate.lib.smtlibutils.polynomials.PolynomialRelation.solveForSubject(PolynomialRelation.java:451)
	at de.uni_freiburg.informatik.ultimate.lib.smtlibutils.polynomials.SolveForSubjectUtils.tryToHandleDivModSubterm(SolveForSubjectUtils.java:212)
	at de.uni_freiburg.informatik.ultimate.lib.smtlibutils.polynomials.SolveForSubjectUtils.findTreatableDivModSubterm(SolveForSubjectUtils.java:585)
	at de.uni_freiburg.informatik.ultimate.lib.smtlibutils.polynomials.SolveForSubjectUtils.findTreatableDivModSubterm(SolveForSubjectUtils.java:573)
	at de.uni_freiburg.informatik.ultimate.lib.smtlibutils.polynomials.SolveForSubjectUtils.solveForSubject(SolveForSubjectUtils.java:76)
	at de.uni_freiburg.informatik.ultimate.lib.smtlibutils.polynomials.PolynomialRelation.solveForSubject(PolynomialRelation.java:451)
	at de.uni_freiburg.informatik.ultimate.lib.smtlibutils.quantifier.DualJunctionDer$DerHelperMcsbr.solveForSubject(DualJunctionDer.java:408)
	at de.uni_freiburg.informatik.ultimate.lib.smtlibutils.quantifier.DualJunctionDer$DerHelperMcsbr.solveForSubject(DualJunctionDer.java:1)
	at de.uni_freiburg.informatik.ultimate.lib.smtlibutils.quantifier.DualJunctionDer$IDerHelper.findBestReplacementSbr(DualJunctionDer.java:312)
	at de.uni_freiburg.informatik.ultimate.lib.smtlibutils.quantifier.DualJunctionDer$IDerHelper.tryToEliminateSbr(DualJunctionDer.java:328)
	at de.uni_freiburg.informatik.ultimate.lib.smtlibutils.quantifier.DualJunctionDer.tryToEliminateOne(DualJunctionDer.java:231)
	at de.uni_freiburg.informatik.ultimate.lib.smtlibutils.quantifier.DualJunctionDer.tryExhaustivelyToEliminate(DualJunctionDer.java:198)
	at de.uni_freiburg.informatik.ultimate.lib.smtlibutils.quantifier.DualJunctionDer.tryToEliminateOne(DualJunctionDer.java:182)
	at de.uni_freiburg.informatik.ultimate.lib.smtlibutils.quantifier.DualJunctionDer.tryExhaustivelyToEliminate(DualJunctionDer.java:149)
	at de.uni_freiburg.informatik.ultimate.lib.smtlibutils.quantifier.DualJunctionDer.tryToEliminate(DualJunctionDer.java:127)
	at de.uni_freiburg.informatik.ultimate.lib.smtlibutils.quantifier.QuantifierPusher.tryToEliminateOne(QuantifierPusher.java:869)
	at de.uni_freiburg.informatik.ultimate.lib.smtlibutils.quantifier.QuantifierPusher.applyNewEliminationTechniquesExhaustively(QuantifierPusher.java:835)
	at de.uni_freiburg.informatik.ultimate.lib.smtlibutils.quantifier.QuantifierPusher.applyDualJunctionEliminationTechniques(QuantifierPusher.java:753)
	at de.uni_freiburg.informatik.ultimate.lib.smtlibutils.quantifier.QuantifierPusher.applyEliminationToAtom(QuantifierPusher.java:272)
	at de.uni_freiburg.informatik.ultimate.lib.smtlibutils.quantifier.QuantifierPusher.convert(QuantifierPusher.java:208)
	at de.uni_freiburg.informatik.ultimate.logic.TermTransformer.cacheConvert(TermTransformer.java:131)
```

**Comment by Heizmann** [comment]
I cannot reproduce this.
I am using 54605efd74 without local changes, solvers provided by the Ultimate repository (with and wihout -ea).
Here, z3 does not terminate.
[log.txt](https://github.com/ultimate-pa/ultimate/files/6410274/log.txt)
Maybe fixed by https://github.com/ultimate-pa/ultimate/commit/ef9aadb27a4a01961104f6cd9a8586130083aafc an a duplicate of #552.

**Comment by danieldietsch** [comment]
Seems fixed, it does not occur on c225d54ecd  (see [overview](https://struebli.informatik.uni-freiburg.de/logs/student-acceleration/20210503-102627-student-acceleration-c225d54ecd/overview))
